test-data/unit/fixtures/dict.pyi

```diff
@@ -11,26 +11,28 @@ class object:
 
 class type: pass
 
-class dict(Mapping[KT, VT], Iterable[KT], Generic[KT, VT]):
+class dict(Generic[KT, VT]):
     @overload
     def __init__(self, **kwargs: VT) -> None: pass
     @overload
     def __init__(self, arg: Iterable[Tuple[KT, VT]], **kwargs: VT) -> None: pass
+    def __getitem__(self, key: KT) -> VT: pass
     def __setitem__(self, k: KT, v: VT) -> None: pass
     def __iter__(self) -> Iterator[KT]: pass
     def update(self, a: Mapping[KT, VT]) -> None: pass
     @overload
     def get(self, k: KT) -> Optional[VT]: pass
     @overload
     def get(self, k: KT, default: Union[KT, T]) -> Union[VT, T]: pass
+    def __len__(self) -> int: ...
 
 class int: # for convenience
     def __add__(self, x: int) -> int: pass
 
 class str: pass # for keyword argument key type
 class unicode: pass # needed for py2 docstrings
 
-class list(Iterable[T], Generic[T]): # needed by some test cases
+class list(Generic[T]): # needed by some test cases
```

Comment:
Because there is an explicit ``__iter__`` defined below. I want to check that it will be picked up correctly by the existing tests. Speaking of this, we could actually remove some explicit bases in typeshed (similarly to what you recently asked concerning ``SupportsInt`` etc.)